sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -760,6 +766,7 @@ def compile(self,
               pipeline_func,
               package_path,
               type_check=True,
+              enable_cache=True,
```

Comment:
If we are going with annotations as flag, maybe we don't need to have this argument over here

Reply:
@Tomcli 
I left this in so caching would be enabled by default, but removing caching from the whole pipeline could be easily done by adding a single flag `--disable-cache` at compile. If we remove this then someone who wants to disable caching would have to add `task1.add_pod_label('pipelines.kubeflow.org/cache_enabled', "true")` to all of their tasks.
If you are saying we should remove enabling caching by default then I can remove all that logic?

Reply:
you can take a look at this example, we added a new way to have annotations shared for the whole pipeline. If someone wants to disable it, they can do it using the tekton_pipeline_conf
https://github.com/kubeflow/kfp-tekton/blob/master/sdk/python/tests/compiler/testdata/tekton_pipeline_conf.py

The original Kubeflow way is to add annotations to each task as well, but just in Argo there's no annotations applied to the whole pipeline because it's not supported.